backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeToolService.java

```diff
@@ -297,9 +297,26 @@ public EdgeTool createFeatureTypingEdgeTool(List<NodeDescription> targetElementD
                 .variableName("newInstance")
                 .children(changeContextNewInstance.build());
 
+        var feedback = this.viewBuilderHelper.newChangeContext()
+                .expression("aql:self.infoMessage('A feature typing already exists between these elements.')")
+                .build();
+
+        var createIfPossible = this.viewBuilderHelper.newIf()
+                .conditionExpression("aql:not hasExistingFeatureTyping")
+                .children(createInstance.build());
+
+        var informOtherwise = this.viewBuilderHelper.newIf()
+                .conditionExpression("aql:hasExistingFeatureTyping")
+                .children(feedback);
+
+        var letExistingFeatureTyping = this.viewBuilderHelper.newLet()
+                .variableName("hasExistingFeatureTyping")
+                .valueExpression("aql:self.eContents(sysml::FeatureTyping).general->includes(semanticEdgeTarget)")
```

Comment:
I have a doubt about the precise condition we should test here, and whether this should be encapsulated inside a service.

Reply:
yes sorry for that, just a typo on my side